pkg/downsample/downsample.go

```diff
@@ -0,0 +1,168 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package downsample
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+
+	"github.com/timescale/promscale/pkg/internal/day"
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/util"
+)
+
+const (
+	setDownsamplingStateSQL       = "SELECT prom_api.set_downsampling_state($1)"
```

Comment:
Frankly having three different SQL functions to update downsampling seems confusing

Reply:
I can hardly tell what is the difference between: `update_downsampling_state` and `set_downsampling_state`. Also there is another function named `create_or_update_downsampling`.

Reply:
If we named the setting `enable_downsampling` instead of `downsample` and then named the functions `set_enable_downsampling` and `get_enable_downsampling` that would clear things up a little IMO